### PR TITLE
[Serve] [Doc] Add basic Serve tutorial

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds a very basic tutorial for Ray Serve for serving a Counter class.  The idea is that this is the first page a brand-new user would go to after landing on the Serve quickstart page.  This tutorial links out to more advanced sections of the Serve docs where relevant.

The existing in-depth tutorials have been renamed "Advanced Tutorials", but perhaps there's a better name for these.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
+==============
+Basic Tutorial
```
Comment (edoakes) [inline doc/source/serve/tutorial.rst:2]: Not sure I like this name, maybe we can call it "End-to-end Tutorial" ?

**Review by edoakes — APPROVED** [review]

File: doc/source/serve/tutorial.rst (hunk 1 above)
Comment (simon-mo) [reply doc/source/serve/tutorial.rst:2]: walkthrough?

**Comment by architkulkarni** [comment]
I renamed it to "End-to-End Tutorial," I didn't have a strong preference.  Docs build passed so I'm marking this as Tests-ok

**Comment by simon-mo** [comment]
if you call this "end-to-end", it should include how to scale backend, delete backend, update backend, and shutdown serve 😉. otherwise it's not end to end. we can do this in a follow up PR or just this PR. up to you.

**Pull request merged**